File: app/src/main/java/knf/kuma/download/DownloadReceiver.java
package knf.kuma.download;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class DownloadReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        int did = intent.getIntExtra("did", 0);
        switch (intent.getIntExtra("action", -1)) {
            case DownloadManager.ACTION_PAUSE:
                DownloadManager.pause(did);
                break;
            case DownloadManager.ACTION_RESUME:
                DownloadManager.resume(did);
                break;
            case DownloadManager.ACTION_CANCEL:
                DownloadManager.cancel(intent.getStringExtra("eid"));
                break;
        }
    }
}

File: app/src/main/java/knf/kuma/download/DownloadReceiver.kt
package knf.kuma.download

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

class DownloadReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val did = intent.getIntExtra("did", 0)
        when (intent.getIntExtra("action", -1)) {
            DownloadManager.ACTION_PAUSE -> DownloadManager.pause(did)
            DownloadManager.ACTION_RESUME -> DownloadManager.resume(did)
            DownloadManager.ACTION_CANCEL -> DownloadManager.cancel(intent.getStringExtra("eid"))
        }
    }
}
